src/android/main.cpp

```diff
@@ -36,20 +37,105 @@ static std::vector<std::string> get_java_args(android_app* pApp)
     return args;
 }
 
+// The Android activity can go by many more states, like PAUSED.
+// Right now, we just need to be able not to crash if we are stopped and resumed.
+enum ApplicationState
+{
+    // The JNI code is loaded and running, but the activity is not started.
+    READY,
+    // The activity is started, the Application/Window code should handle events.
+    RUNNING,
+    // The activity is being destroyed. This is a transient state until we return.
+    DESTROYED
+};
+
+// NOTE: JNI libraries can outlive the activity. Meaning we can re-enter the android_main
+// without re-running global constructors or reseting the BSS.
+static ApplicationState gApplicationState = READY;
+
+// When the activity not running, we have no window. To simplify BigWheels code, we
+// handle those states here, and restart the application once we are back live.
+static void defaultCommandHandler(struct android_app* app, int32_t cmd)
+{
+    switch (cmd) {
+        case APP_CMD_START:
+            gApplicationState = RUNNING;
+            break;
+        case APP_CMD_STOP:
+            gApplicationState = READY;
+            break;
+        case APP_CMD_DESTROY:
+            gApplicationState = DESTROYED;
+            break;
+        case APP_CMD_INIT_WINDOW:
+        case APP_CMD_TERM_WINDOW:
+        case APP_CMD_WINDOW_RESIZED:
+        case APP_CMD_WINDOW_REDRAW_NEEDED:
+        case APP_CMD_CONTENT_RECT_CHANGED:
+            PPX_ASSERT_MSG(false, "Handled in the default-handler a message we shouldn't. This is a bug.");
+            break;
+        default:
+            break;
+    }
+}
+
+int32_t defaultInputHandler(struct android_app*, AInputEvent*)
+{
+    PPX_ASSERT_MSG(false, "Handled an input message without a window. This is a bug.");
+    return 0;
+}
+
+void RunAndroidEventLoop(struct android_app* pApp)
```

Comment:
Are there any cases where this wouldn't use the default command handler? If not, it should be renamed to disambiguate it from the "Android event loop".

Reply:
What do you mean? Like outside of the in-app event loop?
If it was just the "AndroidEventLoop" name which was too generic?
What about "WaitForNonIdleState"? Which is what we do, as the 2 only exists are either running, or the transient exit state.